### DISCUSSION TITLE
Calling AETitle and Remote IP in the OnCFindRequest method SCP

### DISCUSSION THREAD

**Discussion by Ealfaro41** [fo-dicom/fo-dicom#1270] [Q&A] [answered]
Hi, I wanted to ask you if there is any way to get the Calling AETitle and Remote IP inside the method 
[public IEnumerable <DicomCFindResponse> OnCFindRequest (DicomCFindRequest request)] (SCP Provider) 

I'm making a WorkList SCP application and I need this data for various settings and filters, but I can't find a way to get this data from the received parameter. If this is not possible, is there a way to relate the association request of the device to the subsequent call to OnCFindRequest?

**Comment by mrbean-bremen** [comment]
As far as I understand, you can get these from the association in `OnReceiveAssociationRequest(Async)`, e.g. during association negotiation. As this always happens before the actual request, you can get the needed properties there (from the properties `CallingAE` and `RemoteHost` of `DicomAssociation`).

**Reply by Ealfaro41** [reply]
Hello, thanks for your answer, it is correct, but the problem is that I need that data inside the "OnCFindRequest" method and I can't find a way to relate which association (OnReceiveAssociationRequest(Async)) it refers to (OnCFindRequest) call, since there are multiple associations and calls to "OnCFindRequest" and not I can recognize in this method where AETitle came from.

**Reply by Ealfaro41** [reply]
It is good to clarify that I am in an environment where multiple clients are associating and firing the DicomCFindRequest method

**Reply by mrbean-bremen** [reply]
You are right - at that point it is not clear which association it belongs to. It _is_ known at the point where the callback is called, but I guess passing it into the callback as parameter is not wanted... @gofal, do you know of a way to access the association from a callback, or is this something to be considered as a feature?

**Reply by Ealfaro41** [reply]
@gofal do you have any idea how this problem can be solved? I am a bit lost here, from what I have analyzed I can only get this data in the OnReceiveAssociationRequest.

**Comment by Ealfaro41** [comment, marked as answer]
https://github.com/fo-dicom/fo-dicom/issues/968
